### PR TITLE
Clarify #dbRef documentation

### PR DESCRIPTION
**Pull request by kibertoad**
fixes https://github.com/Vincit/objection.js/issues/975

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -10,6 +10,7 @@
     "test-fast": "mocha --slow 10 --timeout 15000 --reporter spec --recursive tests --bail --exclude \"tests/unit/relations/files/**\"",
     "test-opt": "mocha --slow 10 --timeout 15000 --reporter spec --recursive tests --bail --trace_opt --trace_deopt --exclude \"tests/unit/relations/files/**\"",
     "test-prof": "mocha --slow 10 --timeout 15000 --reporter spec --recursive tests --prof --exclude \"tests/unit/relations/files/**\"",
+    "test-sqlite": "cross-env DATABASES=sqlite3 mocha --slow 10 --timeout 15000 --reporter spec --recursive tests --exclude \"tests/unit/relations/files/**\"",
```
Comment (kibertoad) [inline package.json]: This was added for the convenience of local debugging, more often than not sqlite is quite enough.

File: tests/integration/upsertGraph.js [hunk 2]
```diff
@@ -545,6 +548,31 @@ module.exports = session => {
...
 
+    it('does not support #dbref for upserts', () => {
+      const upsert = {
+        id: 2,
+        model1Id: 3,
+        model1Relation2: [
+          {
+            '#dbRef': 1,
+            model2Prop1: 'updated hasMany 1'
+          },
```
[the remark this answers is not in the document]
Comment (kibertoad) [reply tests/integration/upsertGraph.js]: Thanks for clarifying that! I probably should try to fix it, not sure how complicated that part of the code is.

[the remark this answers is not in the document]
Comment (koskimas) [reply tests/integration/upsertGraph.js]: Don't bother. I'm in the process of rewriting the `upsertGraph` and `insertGraph` stuff. It's unmaintainable at the moment.

File: doc/index.md [hunk 3]
```diff
@@ -1554,6 +1555,8 @@ await Person
   }, options);
 ```
 
+> Note that `upsertGraph` does not support the parameter `#dbRef` (which is supported by `insertGraph`). Use `relate` option and pass an id as depicted above instead.
+
```
[the remark this answers is not in the document]
Comment (koskimas) [reply doc/index.md]: This is now fixed in master. You can use #dbRef

File: tests/integration/upsertGraph.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (koskimas) [reply tests/integration/upsertGraph.js]: This is now fixed in master.

[the remark this answers is not in the document]
Comment (kibertoad) [reply tests/integration/upsertGraph.js]: Fixed documentation.

File: tests/integration/upsertGraph.js [hunk 4]
```diff
@@ -1,11 +1,14 @@
 const raw = require('../../').raw;
 const expect = require('expect.js');
 const chai = require('chai');
+const chaiAsPromised = require('chai-as-promised');
 const Promise = require('bluebird');
 const transaction = require('../../').transaction;
 const ValidationError = require('../../').ValidationError;
 const mockKnexFactory = require('../../testUtils/mockKnex');
 
+chai.use(chaiAsPromised);
```
Comment (koskimas) [inline tests/integration/upsertGraph.js]: This isn't used anywhere right? Why did you add it?

File: package.json [hunk 5]
```diff
@@ -70,8 +71,10 @@
     "@types/node": "^10.0.6",
     "babel-eslint": "^10.0.1",
     "chai": "^4.2.0",
+    "chai-as-promised": "^7.1.1",
```
Comment (koskimas) [inline package.json]: This is not used anywhere as far as I can tell

**Comment by koskimas** [comment]
You still need to manually rebase this and also the tests seem to fail

File: tests/integration/upsertGraph.js (hunk 4 above)
Comment (kibertoad) [reply tests/integration/upsertGraph.js]: yup, not needed anymore, was needed for promise rejection assertion

**Comment by kibertoad** [comment]
@koskimas What are the benefits of manual rebasing as opposed to doing automatic GitHub rebasing or squashing on merge?
(and yeah, need to improve the way assertion is done)

**Comment by kibertoad** [comment]
Actually, new test seems redundant now, there is already an existing one that uses dbRef with upserts, so I've removed it.

**Comment by koskimas** [comment]
@kibertoad 
> What are the benefits of manual rebasing as opposed to doing automatic GitHub rebasing or squashing on merge

Github doesn't solve conflicts

**Comment by kibertoad** [comment]
@koskimas "This branch has no conflicts with the base branch". What am I missing :)?

**Comment by koskimas** [comment]
I don't know :smile:

**Comment by kibertoad** [comment]
Superseded by https://github.com/Vincit/objection.js/pull/1195
